### ISSUE TITLE
wrong_self_convention cannot be allowed at the fn level

### ISSUE THREAD

**Issue by birkenfeld** [Manishearth/rust-clippy#1137]
Example code:
```rust
struct X;
impl X {
    fn from_foo(self) {}
}
fn main() { }
```
This warns because `from_foo` should not take `self`.

I can suppress it for the whole impl by placing `#[allow(wrong_self_convention)]` before `impl X`, but currently it's impossible to suppress this warning (more naturally) at the fn level, i.e. with `#[allow(wrong_self_convention)]` before `fn from_foo`.
